components/base/sensorcontrolled/spin.go

```diff
@@ -172,6 +172,23 @@ func (sb *sensorBase) stopSpinWithSensor(
 func getTurnState(currYaw, startYaw, targetYaw, dir, angleDeg, errorBound float64) (atTarget, overShot, minTravel bool) {
 	atTarget = math.Abs(targetYaw-currYaw) < errorBound
 	overShot = hasOverShot(currYaw, startYaw, targetYaw, dir)
+	if dir == 1 {
+		offset := 0.0
+		if startYaw-boundCheckOverShot < 0 {
+			offset = -360.0
+		}
+		if currYaw < startYaw && currYaw+offset > startYaw-boundCheckOverShot {
+			overShot = false
+		}
+	} else {
+		offset := 0.0
+		if startYaw+boundCheckOverShot > 0 {
+			offset = 360.0
+		}
+		if currYaw > startYaw && currYaw+offset < startYaw+boundCheckOverShot {
+			overShot = false
+		}
+	}
```

Comment:
not confident this is working still need to do some more testing

Reply:
also the comments might be a bit overkill and/or not make sense, let me know. I think this was kind of a confusing way to go about it